clang/include/clang/CIR/Common/ABIArgInfo.h

```diff
@@ -0,0 +1,226 @@
+//==-- ABIArgInfo.h - Abstract info regarding ABI-specific arguments -------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Defines ABIArgInfo and associated types used by CIR to track information
+// regarding ABI-coerced types for function arguments and return values. This
+// was moved to the common library as it might be used by both CIRGen and
+// passes.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIR_COMMON_ABIARGINFO_H
+#define CIR_COMMON_ABIARGINFO_H
+
+#include "mlir/IR/Types.h"
+#include "clang/AST/Type.h"
+#include <cstdint>
+
+namespace cir {
+
+/// ABIArgInfo - Helper class to encapsulate information about how a specific C
```

Comment:
Delete `ABIArgInfo - ` and elsewhere you do copy-n-paste, we're gonna get asked to remove this when upstreaming, so just do it now plz